src/rpm/headers/types.rs

```diff
@@ -201,9 +173,37 @@ pub struct FileOptions {
     pub(crate) caps: Option<FileCaps>,
 }
 
+// @todo: should we even have "default permissions" / mode, or use Option<FileMode>?
+// if they can be skipped (unsure), 'inherit_permissions' could go away
 impl FileOptions {
+    /// Create a new FileOptions for a regular file
+    #[allow(clippy::new_ret_no_self)]
+    pub fn regular(dest: impl Into<String>) -> FileOptionsBuilder {
+        FileOptionsBuilder {
+            inner: FileOptions {
+                destination: dest.into(),
+                user: "root".to_string(),
+                group: "root".to_string(),
+                symlink: "".to_string(),
+                mode: FileMode::regular(0o664),
+                flag: FileFlags::empty(),
+                inherit_permissions: true,
+                caps: None,
+            },
+        }
+    }
+
+    /// Create a new FileOptions for a directory
+    ///
+    /// Used to explicitly own the directory itself but not it's contents. This does NOT
+    /// process any of the contents of a directory.
     #[allow(clippy::new_ret_no_self)]
-    pub fn new(dest: impl Into<String>) -> FileOptionsBuilder {
+    pub fn dir(dest: impl Into<String>) -> FileOptionsBuilder {
+        // @todo: is this problematic? `with_file` has a source argument, but I do not believe %dir uses
+        // any "source" necessarily, just a destination.  Maybe we need `PackageBuilder::with_dir()`,
+        // except that sounds like it would process files recursively inside a directory, and we would
+        // still need to duplicate much of this code.
```

Comment:
open questions

Reply:
On second thought, in case we don't "inherit permissions" and there is no default, it would probably need to stat the directory to get those permissions on its own?